lib/xen.pm

```diff
@@ -0,0 +1,67 @@
+# XEN regression tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper package for common xen operations
+# Maintainer: Jan Baier <jbaier@suse.cz>
+
+package xen;
+
+use base 'consoletest';
+use strict;
+use testapi;
+use utils;
+
+# Supported guest configuration
+#   * location of the installation tree
+#   * autoyast profile
+#   * extra parameters for virsh create / xl create
+our %guests = (
+    A_sles12_PV => {
+        autoyast     => 'autoyast_xen/xen-SLES12-SP3-PV.xml',
+        extra_params => '--paravirt',
+        location     => 'http://mirror.suse.cz/install/SLP/SLE-12-SP3-Server-GM/x86_64/DVD1/',
+    },
+    A_sles12_HVM => {
+        autoyast     => 'autoyast_xen/xen-SLES12-SP3-FV.xml',
+        extra_params => '--hvm',
+        location     => 'http://mirror.suse.cz/install/SLP/SLE-12-SP3-Server-GM/x86_64/DVD1/',
```

Comment:
As it's same url, I would keep in a single place. Also, would it make sense to simply inject `VERSION` variable here, so we don't have to rewrite this after new SP is out?
Potentially, all of these could be test suite settings, but that's matter of taste.

Reply:
In fact, the key should be ``A_sles12_sp3_HVM``, after new SP is out, we should add new key and config if we want to cover that SP as well.